v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/stack-traces/stack-trace-solidity-errors.ts

```diff
@@ -181,7 +181,7 @@ function sourceReferenceToSolidityCallsite(
   );
 }
 
-function getMessageFromLastStackTraceEntry(
+export function getMessageFromLastStackTraceEntry(
```

Comment:
Actually, reusing this adds a ton of network-simulation-level concepts to the error messages. For example, `"Transaction reverted: library was called directly"`, while there's no concept of transactions here. It also adds a ton of noise that's only present for backwards compatibility with existing libraries, like `"VM Exception while processing transaction:"`.

Ideally we'd have a cleaner version of the messages in the solidity tests reporter. 

I guess the simpler change, at least for now, would be duplicating this function in this same file (with a different name) and polishing the error messages a bit.

`encodeStackTraceEntry` doesn't seem to have this problem.

Reply:
That's a great point! Actually, revisiting this piece of code, I started thinking if we even have to try and retrieve the message from the traces at all. My reasoning is that we might be getting the same information via `TestResult.reason` already anyway. 

If my thinking is correct (which I'm yet to double-check), then we could do [`5a25255` (#6235)](https://github.com/NomicFoundation/hardhat/pull/6235/commits/5a25255e26d98c950d519238919f8d0a7a6d4567) which would result in this example output:

<img width="493" alt="Screenshot 2025-02-07 at 16 54 56" src="https://github.com/user-attachments/assets/f178a6ec-9ea7-478a-bb37-1f4432955cda" />

Reply:
I think that looks good

Reply:
I'm checking in with Agost about this to make sure. I'll report our findings back here.

Reply:
Agost is the best! So it seems `forge` uses `TestResult.reason` as their main message for the test outputs, which tells me it should contain all the crucial information we need.

I'd say, let's go with this, and if we find ourselves lacking detail at some point, we can look again if we can find it somewhere among trace props.